.github/workflows/build-cosi.yaml

```diff
@@ -0,0 +1,55 @@
+---
+name: COSI Builds
+
+on:
+  schedule:
+    - cron: '0 0 * * 0'  # Every sunday night midnight
+  pull_request:
+  workflow_dispatch:   # Or manually - for testing
+
+jobs:
+  build-cosi:
+    runs-on: self-hosted
```

Comment:
If the build isn't particularly resource intensive, it's better to use `ubuntu-latest` as then it will run on GitHub's workers.